atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -358,21 +358,47 @@ def run_image_task(self, image_build_conf):
     def stream_filesystem(self, task_id, filesystem_regex):
         filesystem = self.download_filesystem(task_id, filesystem_regex)
 
-        new_base_image = self.import_base_image(filesystem)
-        self.workflow.builder.set_base_image(new_base_image)
-        defer_removal(self.workflow, new_base_image)
+        new_parent_image = self.import_base_image(filesystem)
+        new_imagename = ImageName.parse(new_parent_image)
 
-        return new_base_image
+        if self.workflow.builder.custom_base_image:
+            self.workflow.builder.set_base_image(new_imagename)
 
-    def run(self):
+        # setting the same new image for all custom base images
```

Comment:
Is that right though?

Reply:
removed comment and changing just 1st, since it will raise error in build.py if there are 2+ different ones